File: 期末考/Q6 Python繼承/Sphere.py
from Shape import Shape
class Sphere(Shape):
    def __init__(self, newRadius, newColor):
        self.__radius = newRadius
        super().__init__(newColor) #繼承別人的變數用super
    def setRadius(self, newRadius):
        self.__radius = newRadius
    def getRadius(self):
        return self.__radius
    def getVolume(self):
        return 3.14159 * self.getRadius() * self.getRadius() * self.getRadius() * 4 / 3
    def __str__(self):
        return "Sphere"

File: 期末考/Q2 C++繼承/Sphere.cpp
#include"Sphere.h"
Sphere::Sphere(double newRadius, string newColor)
{
    setRadius(newRadius);
    setColor(newColor);
}
void Sphere::setRadius(double newRadius)
{
    radius = newRadius;
}
double Sphere::getRadius()
{
    return radius;
}
double Sphere::getVolume()
{
    return 4 * 3.14159 * radius * radius * radius / 3.0;
}
void Sphere::toString()
{
    cout << "Sphere:";
}